src/seeds.history.cpp

```diff
@@ -113,202 +126,236 @@ void history::trxentry(name from, name to, asset quantity) {
       });
+    }
   }
 
-  if (!from_is_organization) {
-    // delayed update
-    cancel_deferred(from.value);
+  cancel_deferred(from.value);
 
-    // delayed update
-    cancel_deferred(from.value);
+  action a(
+    permission_level{contracts::history, "active"_n},
+    get_self(),
+    "savepoints"_n,
+    std::make_tuple(transaction_id, timestamp)
+  );
 
-    action a(
-        permission_level{contracts::harvest, "active"_n},
-        contracts::harvest,
-        "updatetxpt"_n,
-        std::make_tuple(from)
-    );
+  transaction tx;
+  tx.actions.emplace_back(a);
+  tx.delay_sec = 1;
+  tx.send(from.value, _self);
+}
 
-    transaction tx;
-    tx.actions.emplace_back(a);
-    tx.delay_sec = 1; 
-    tx.send(from.value, _self);
-  }
 
-}
+void history::savepoints(uint64_t id, uint64_t timestamp) {
+  require_auth(get_self());
 
-// return true if anything was cleaned up, false otherwise
-bool history::clean_old_tx(name org, uint64_t chunksize) {
-  
-  auto now = eosio::current_time_point().sec_since_epoch();
-  auto cutoffdate = now - utils::moon_cycle * 3;
+  auto date = eosio::time_point_sec(timestamp / 86400 * 86400);
+  uint64_t day = date.utc_seconds;
 
-  org_tx_tables orgtx(get_self(), org.value);
-  uint64_t count = 0;
-  auto transactions_by_time = orgtx.get_index<"bytimestamp"_n>();
-  auto tx_time_itr = transactions_by_time.begin();
-  bool result = false;
-  while(tx_time_itr != transactions_by_time.end() && tx_time_itr->timestamp < cutoffdate && count < chunksize) {
-    tx_time_itr = transactions_by_time.erase(tx_time_itr);
-    result = true;
-    count++;
-  }  
-  return result;
-  
-}
+  daily_transactions_tables transactions(get_self(), day);
+  auto transactions_by_from_to = transactions.get_index<"byfromto"_n>();
 
-void history::orgtxpoints(name org) {
-  require_auth(get_self());
-  orgtxpt(org, 0, 200, 0);
-}
+  auto titr = transactions.find(id);
+  check(titr != transactions.end(), "transaction not found");
+  name from = titr -> from;
+  name to = titr -> to;
 
-void history::orgtxpt(name org, uint128_t start_val, uint64_t chunksize, uint64_t running_total) {
-  require_auth(get_self());
+  auto uitr_from = users.find(from.value);
+  auto uitr_to = users.find(to.value);
 
-  org_tx_tables orgtx(get_self(), org.value);
+  uint64_t max_number_transactions = config_get("htry.trx.max"_n);
+
+  uint128_t from_to_id = (uint128_t(titr -> from.value) << 64) + titr -> to.value;
   uint64_t count = 0;
 
-  if (start_val == 0 && clean_old_tx(org, chunksize)) {    // Step 0 - remove all old transactions
-    fire_orgtx_calc(org, 0, chunksize, running_total);
-    return;
+  auto ft_itr = transactions_by_from_to.find(from_to_id);
+  auto current_itr = ft_itr;
+
+  while (ft_itr != transactions_by_from_to.end() && 
+      count <= max_number_transactions && 
+      ft_itr -> from == from && ft_itr -> to == to) {
+
+    if (ft_itr -> volume < current_itr -> volume) {
+      current_itr = ft_itr;
```

Comment:
current_itr is set to the smallest of the - potentially - 3 transactions

Reply:
Yes, I you said below, it erases the smallest so it keeps the 2 greatest qualifying transactions. I did it that way because I didn't want to determine that in the `trxentry` action due to the reliability concerns, that's why it adds the transaction to the table and then in `savepoints` checks if it should be kept.